tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -424,11 +365,86 @@ def assertAllBackendsReceivedRpcs(self, lb_stats):
                 msg=f'Backend {backend} did not receive a single RPC')
 
 
-class TdPropagationRetryableError(Exception):
-    pass
+class XdsKubernetesIsolatedTestCase(XdsKubernetesBaseTestCase,
+                                    metaclass=abc.ABCMeta):
+    """Isolated test case.
+
+    Base class for tests cases where infra resources are created before
+    each test, and destroyed after.
```

Comment:
Should we talk more about what kind of resources will be recreated? For example, client pods, server pods, GCP resources.

Reply:
Most of it  is delegated to the classes to implement, see `initTrafficDirectorManager`, `initKubernetesServerRunner`, `initKubernetesClientRunner`. I think for the rest the code is self-documenting enough.

Reply:
Just realized these are just inits.

Reply:
On another hand, the test itself controls what's created. This class mostly does the cleanup, and provides some nonce for the resources.